### DISCUSSION TITLE
Not able to connect minio bucket from hive metastore

### DISCUSSION THREAD

**Discussion by mnhaque** [minio/minio#15305] [Q&A] [answered]
Getting below error while trying to connect minio tenant bucket from hive metastore
`MetaException(message:Got exception: org.apache.hadoop.fs.s3a.AWSBadRequestException doesBucketExist on minio.minio-tenant-ns.svc.cluster.local: com.amazonaws.services.s3.model.AmazonS3Exception: Bad Request (Service: Amazon S3; Status Code: 400; Error Code: 400 Bad Request; Request ID: null; S3 Extended Request ID: null), S3 Extended Request ID: null:400 Bad Request: Bad Request (Service: Amazon S3; Status Code: 400; Error Code: 400 Bad Request; Request ID: null; S3 Extended Request ID: null))
        at org.apache.hadoop.hive.metastore.RetryingHMSHandler.<init>(RetryingHMSHandler.java:84)
        at org.apache.hadoop.hive.metastore.RetryingHMSHandler.getProxy(RetryingHMSHandler.java:93)
        at org.apache.hadoop.hive.metastore.HiveMetaStore.newRetryingHMSHandler(HiveMetaStore.java:8541)
        at org.apache.hadoop.hive.metastore.HiveMetaStore.newRetryingHMSHandler(HiveMetaStore.java:8536)
        at org.apache.hadoop.hive.metastore.HiveMetaStore.startMetaStore(HiveMetaStore.java:8806)
        at org.apache.hadoop.hive.metastore.HiveMetaStore.main(HiveMetaStore.java:8723)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.hadoop.util.RunJar.run(RunJar.java:323)
        at org.apache.hadoop.util.RunJar.main(RunJar.java:236)`

Minio deployment details
![image](https://user-images.githubusercontent.com/9695218/179155826-6fcdec4a-01dc-4e02-a5ed-b8fbc7b0dc1e.png)
![image (1)](https://user-images.githubusercontent.com/9695218/179155839-46a6d2cc-f794-4418-a784-8400e10937cd.png)

Hive metastore configuration details
`<property>
        <name>fs.defaultFS</name>
        <value>s3a://minio.minio-tenant-ns.svc.cluster.local:443</value>
    </property>
    <property>
        <name>fs.s3a.connection.ssl.enabled</name>
        <value>true</value>
    </property>
    <property>
        <name>fs.s3a.endpoint</name>
        <value>http://minio.minio-tenant-ns.svc.cluster.local:443/</value>
    </property>
    <property>
        <name>fs.s3a.access.key</name>
        <value>miniouser</value>
    </property>
    <property>
        <name>fs.s3a.secret.key</name>
        <value>minio123</value>
    </property>
    <property>
        <name>fs.s3a.path.style.access</name>
        <value>true</value>
    </property>
    <property>
        <name>fs.s3a.impl</name>
        <value>org.apache.hadoop.fs.s3a.S3AFileSystem</value>
    </property><property>
        <name>fs.defaultFS</name>
        <value>s3a://minio.minio-tenant-ns.svc.cluster.local:443</value>
    </property>
    <property>
        <name>fs.s3a.connection.ssl.enabled</name>
        <value>true</value>
    </property>
    <property>
        <name>fs.s3a.endpoint</name>
        <value>http://minio.minio-tenant-ns.svc.cluster.local:443/</value>
    </property>
    <property>
        <name>fs.s3a.access.key</name>
        <value>miniouser</value>
    </property>
    <property>
        <name>fs.s3a.secret.key</name>
        <value>minio123</value>
    </property>
    <property>
        <name>fs.s3a.path.style.access</name>
        <value>true</value>
    </property>
    <property>
        <name>fs.s3a.impl</name>
        <value>org.apache.hadoop.fs.s3a.S3AFileSystem</value>
    </property>`

**Comment by harshavardhana** [comment, marked as answer]
Your server is on TLS, you are trying to connect to MinIO over non-TLS

**Reply by mnhaque** [reply]
@harshavardhana you mean to say the source application needs to be configured with TLS?

**Reply by mnhaque** [reply]
@harshavardhana, can you please provide a little more info. It is not clear to me

**Reply by mnhaque** [reply]
It worked after turning off TLS for the tenant and making related changes in hive metastore.

@harshavardhana is there a way to have TLS configured and to be able to connect from other applications

**Reply by harshavardhana** [reply]
you need to deploy MinIO with TLS enabled

**Reply by mnhaque** [reply]
@harshavardhana 
If we deploy MinIO with TLS enabled, we are not able to connect from hive metastore.
We will be getting the error I mentioned in the beginning.

**Reply by harshavardhana** [reply]
For that you need to trust the TLS certs on MinIO to hive meta store
